### ISSUE TITLE
D loss 'nan'

### ISSUE THREAD

**Issue by liulizhou** [LeeDoYup/DeblurGAN-tf#11]
Discriminator loss nan, why?
```python
[2018-11-22 16:53:22,014 INFO] 0 epoch,  152  batch, Discriminator  Loss:  11.182838
[2018-11-22 16:53:22,109 INFO] [!] Generator Optimization Start
[2018-11-22 16:53:22,978 INFO] [!] Discriminator Optimization Start
[2018-11-22 16:53:23,113 INFO] 0 epoch,  153  batch, Discriminator  Loss:  nan
nan
[2018-11-22 16:53:23,213 INFO] [!] Generator Optimization Start
[2018-11-22 16:53:24,131 INFO] [!] Discriminator Optimization Start
nan
[2018-11-22 16:53:24,265 INFO] 0 epoch,  154  batch, Discriminator  Loss:  nan
[2018-11-22 16:53:24,366 INFO] [!] Generator Optimization Start
[2018-11-22 16:53:24,435 INFO] 0 epoch,  155 batch, Generator Loss:  nan, add loss: nan, perceptual_loss: 0.003620
[2018-11-22 16:53:25,444 INFO] [!] Discriminator Optimization Start
[2018-11-22 16:53:25,580 INFO] 0 epoch,  155  batch, Discriminator  Loss:  nan
nan
[2018-11-22 16:53:25,685 INFO] [!] Generator Optimization Start
[2018-11-22 16:53:26,591 INFO] [!] Discriminator Optimization Start
nan
```

**Comment by LeeDoYup** [comment]
This codes have some problems in training the model.
I will fix them ASAP if i can. Welcome the pull request.

**Comment by liulizhou** [comment]
I have find the problem.
```
tf.sqrt(tf.reduce_sum(tf.square(grad_d_x_hat), reduction_indices=red_idx))
```
the value tf.reduce_sum(tf.square(grad_d_x_hat), reduction_indices=red_idx) may < 0. I add 1e-8 and it is correct. It is like this:
```
slopes = tf.sqrt(1e-8 + tf.reduce_sum(tf.square(grad_d_x_hat), reduction_indices=red_idx))
```

**Comment by LeeDoYup** [comment]
Thank you for your kindness. 
In the code, there was memory leak problem but i fix it and refactory the codes,.
